config/config.go

```diff
@@ -51,7 +53,9 @@ type Context struct {
 	// - New() function: config/load.go
 
 	// HTTPClient is filled in by calling Initialize()
-	HTTPClient *http.Client
+	HTTPClient connect.HTTPClient
+	// HTTPClientBuilder is filled in by calling Initialize() and can be used when the transport needs to be modified
+	HTTPClientBuilder func(transport http.RoundTripper) connect.HTTPClient
```

Comment:
This looks like it adds a fair bit of complexity compared to the previous implementation - instead of adding `HTTPClientBuilder`, we should be able to simply stack all of the transports together in `Initialize` and construct one http.Client consisting of the ErrorHandlingTransport + the oauth2 transport.